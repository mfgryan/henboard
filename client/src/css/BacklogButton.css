.BacklogButton {
    visibility: hidden;
}

.Item:hover > .BacklogButton {
    visibility: visible;
}
