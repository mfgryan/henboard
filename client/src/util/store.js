// redux dep
import { createStore, applyMiddleware } from "redux";
import { logger, validate } from "../util/middleware.js";
import data from "../util/data.js";
import rootReducer from "../reducers/index.js";

const store = createStore(rootReducer,applyMiddleware(logger,validate));

export { store }
