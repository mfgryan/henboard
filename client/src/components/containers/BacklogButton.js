    // redux dep
import React from "react";
import { connect } from "react-redux";
import { Button, Col } from "react-bootstrap";
import { moveItem } from "../../actions/items.js";
import projects from "../../models/projects";
import sprints from "../../models/sprints";

import "../../css/BacklogButton.css";

export const BacklogButton = ({ title, item, buttonClicked }) => {
    return (
        <Col md={4} mdOffset={7} className="BacklogButton">
            <Button onClick={event => buttonClicked(event, item)}>
                {title}
            </Button>
        </Col>
    );
};

const mapStateToProps = (state, ownProps) => {
    let project = projects.getCurrentProject(state);
    let sprint = sprints.getCurrentSprint(state, project.project);
    let newItem = Object.assign({}, ownProps.item, {
        week: sprint.week,
        column: "Todo"
    });
    return {
        title: "Send To " + sprint.week,
        item: newItem
    };
};

const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        buttonClicked: (event, item) => {
            dispatch(moveItem(item));
        }
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(BacklogButton);
