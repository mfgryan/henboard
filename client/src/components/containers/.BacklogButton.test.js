import React from "react";
import BacklogButton from "./BacklogButton";
import renderer from "react-test-renderer";

// redux dep
import { Provider } from "react-redux";
import { store } from "../../util/store.js";

// router dep
import { BrowserRouter as Router } from "react-router-dom";

it('BacklogButton renders correctly', () => {
    let item = {
        project: "henboard",  
        name: "foo",  
        column: "Backlog",  
        value: "bar",  
        week: "01/01/01",  
        openInfo: false,  
        addItem: false
    };
    const tree = renderer.create(
        <Router>
            <Provider store={store}>
                <BacklogButton project="henboard" item={item} />
            </Provider>
        </Router>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
