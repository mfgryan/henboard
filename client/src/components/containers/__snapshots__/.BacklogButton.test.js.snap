exports[`test BacklogButton renders correctly 1`] = `
<div
  className="BacklogButton col-md-4 col-md-offset-7">
  <button
    className="btn btn-default"
    disabled={false}
    onClick={[Function]}
    type="button">
    Send To 05/15/17
  </button>
</div>
`;
